### PR TITLE
Clear fields when check/uncheck conditional in formatters

### LINKED ISSUE #4784: Conditional format not being removed from xml
**Describe the bug**
When you set a conditional format for a table and then if you remove it and set a non conditional format the xml still shows the conditional format and running a query gives an error.  

**To Reproduce**

1. Go to app resources-> record formatters (using visual editor) 
2. Click on accesssion 
3. Add a conditional format (having at least 2 definitions) 
4. Save 
5. Remove conditional format 
6. Add just one field 
7. Save 
8. Check xml to see it still shows the conditional format 
9. Query and see error 

**Expected behavior**
The xml should remove the old conditional format and correct to the new table format. 

**Screenshots**

https://github.com/specify/specify7/assets/150940840/3fbda393-b601-4873-a39c-f99045059719

**Crash Report**

[Specify 7 Crash Report - 2024-04-12T16_49_06.763Z.txt](https://github.com/specify/specify7/files/14961421/Specify.7.Crash.Report.-.2024-04-12T16_49_06.763Z.txt)

Please, also fill out the following information manually:

- OS: Windows 11
- Browser: Chrome 
- Specify 7 Version: edge
- Database Name: lsumzmammals9823
- Collection name: mammals
- User Name: SpAdmin

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4784

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone

### Testing instructions

Go to app resources-> record formatters (using visual editor)
Click on accesssion
Add a conditional format
Save
Reopen
Remove conditional format
- [ ] verify that when you switch between Conditional Format checked and unchecked, the fields are being deleted from the editor (like starting a new formatter from scratch) 
Click on accesssion
Uncheck conditional 
- [ ] verify that when you switch, the fields are being deleted from the editor (like starting a new formatter from scratch) 
- [ ] Check in xml viewer that only the field is present when only field in visual editor 
- [ ] Check in xml viewer that conditionals is/are present when defined in visual editor 
In conclusion, when switching from Conditional Format checked/unchecked the part with the fields should be cleared.

### CODE REVIEW

**Review by emenslin — APPROVED** [review]
### Testing instructions

Go to app resources-> record formatters (using visual editor)
Click on accesssion
Add a conditional format
Save
Reopen
Remove conditional format
- [x] verify that when you switch between Conditional Format checked and unchecked, the fields are being deleted from the editor (like starting a new formatter from scratch) 
Click on accesssion
Uncheck conditional 
- [x] verify that when you switch, the fields are being deleted from the editor (like starting a new formatter from scratch) 
- [x] Check in xml viewer that only the field is present when only field in visual editor 
- [x] Check in xml viewer that conditionals is/are present when defined in visual editor 
In conclusion, when switching from Conditional Format checked/unchecked the part with the fields should be cleared. 

Looks good, I like this behavior a lot!

**Review by melton-jason — APPROVED** [review]
> If the user checks/unchecks "Conditional Format", we need to clear all fields as if no definitions were created. Only "Name", "Title", and "Default" should be preserved when it is cleared.

https://github.com/specify/specify7/issues/4784#issuecomment-2052271775

Looks good with this behavior in mind!

**Review by pashiav — APPROVED** [review]
### Testing instructions

Go to app resources-> record formatters (using visual editor)
Click on accesssion
Add a conditional format
Save
Reopen
Remove conditional format
- [X] verify that when you switch between Conditional Format checked and unchecked, the fields are being deleted from the editor (like starting a new formatter from scratch) 
Click on accesssion
Uncheck conditional 
- [X] verify that when you switch, the fields are being deleted from the editor (like starting a new formatter from scratch) 
- [X] Check in xml viewer that only the field is present when only field in visual editor 
- [X] Check in xml viewer that conditionals is/are present when defined in visual editor 
In conclusion, when switching from Conditional Format checked/unchecked the part with the fields should be cleared. 

---
Looks good! Working as expected.
